File: 240410/줄 세우기/line-up-students.py
N = int(input())
students_input = [list(map(int, input().split()))for _ in range(N)]

students = []
for student_idx, (height, weight) in enumerate(students_input, 1):
    students.append([height, weight, student_idx])

students.sort(key = lambda x: (-x[0], -x[1], -x[2]))
for student in students:
    print(*student)

File: 260104/줄 세우기/line-up-students.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

struct Student 
{
    int height;
    int weight;
    int idx;
};

int main() 
{
    int N;
    cin >> N;

    vector<Student> students;
    students.reserve(N);

    for (int i = 1; i <= N; i++) 
    {
        int h, w;
        cin >> h >> w;
        students.push_back({h, w, i});
    }

    sort(students.begin(), students.end(), [](const Student& a, const Student& b) 
    {
        if (a.height != b.height) return a.height > b.height;  
        if (a.weight != b.weight) return a.weight > b.weight;  
        return a.idx < b.idx;                                  
    });

    for (const auto& s : students) 
    {
        cout << s.height << " " << s.weight << " " << s.idx << "\n";
    }

    return 0;
}
